packages/go_router/tool/run_tests.dart

```diff
@@ -9,25 +9,103 @@
 // ignore_for_file: avoid_print
 
 import 'dart:io';
+import 'package:io/io.dart' as io;
 import 'package:path/path.dart' as p;
+import 'package:yaml_edit/yaml_edit.dart';
 
 Future<void> main(List<String> args) async {
-  if (!Platform.isMacOS) {
-    print('This test can only be run on macOS.');
+  if (!Platform.isMacOS && !Platform.isWindows) {
+    print('This test can only be run on macOS or windows.');
     exit(0);
   }
+  final p.Context ctx = p.context;
   final Directory packageRoot =
-      Directory(p.dirname(Platform.script.path)).parent;
+      Directory(ctx.dirname(_ensureTrimLeadingSeparator(Platform.script.path)))
+          .parent;
+
+  // Copy test_fixes/ to be tested in a temp directory.
+  // This ensures the dart fix can be applied to projects
+  // outside of this package.
+  final Directory testFixesTargetDir = await Directory.systemTemp.createTemp();
+
+  await _prepareTemplate(
+    ctx: ctx,
+    testFixesDir: ctx.join(packageRoot.path, 'test_fixes'),
+    testFixesTargetDir: testFixesTargetDir.path,
+  );
+  final int pubGet = await _runProcess(
+    'dart',
+    <String>[
+      'pub',
+      'upgrade',
+    ],
+    workingDirectory: testFixesTargetDir.path,
+  );
+  if (pubGet != 0) {
+    await cleanUpAndExit(statusCode: pubGet, toDelete: testFixesTargetDir);
+  }
   final int status = await _runProcess(
     'dart',
     <String>[
       'fix',
       '--compare-to-golden',
     ],
-    workingDirectory: p.join(packageRoot.path, 'test_fixes'),
+    workingDirectory: testFixesTargetDir.path,
   );
+  if (status != 0) {
+    await cleanUpAndExit(statusCode: status, toDelete: testFixesTargetDir);
+  }
+  await cleanUpAndExit(statusCode: 0, toDelete: testFixesTargetDir);
```

Comment:
What is the purpose of extracting this method just to call it twice in exactly the same way (since you are passing `0` iff `status` is `0`, making it equivalent to the call above)? You could replace these three lines with the two lines of the method body, and it would be both less code and clearer.